src/Store.js

```diff
@@ -176,122 +239,133 @@ class Store {
+      console.warn(
+        'DeprecationWarning: using `fromServer` options is deprecated and will be removed - use getOne, fetchOne, or findOne.'
+      )
+    }
 
     if (fromServer === true) {
-      // If fromServer is true always fetch the data and return
-      return this.fetchOne(type, id, queryParams)
+      return this.fetchOne(type, id, options)
     } else if (fromServer === false) {
-      // If fromServer is false never fetch the data and return
-      return this.getRecord(type, id, queryParams)
+      return this.getOne(type, id, options)
+    }
+
+    const record = this.getOne(type, id, options)
+    if (record?.id) {
+      return record
     } else {
-      return this.findOrFetchOne(type, id, queryParams)
+      return this.fetchOne(type, id, options)
     }
   }
 
   /**
-   * returns cache if exists, returns promise if not
+   * Get all records with the given `type` and `ids` from the store. This will never fetch from the server.
    *
-   * @method findOrFetchOne
-   * @param {String} type record type
-   * @param id
-   * @param {Object} queryParams will inform whether to return cached or fetch
+   * @method getMany
+   * @param {String} type the type to get
+   * @param {String} ids the ids of the records to get
+   * @param {Object} options { queryParams }
+   * @return {Array} array of records
    */
-  findOrFetchOne = (type, id, queryParams) => {
-    // Get the matching record
-    const record = this.getMatchingRecord(type, id, queryParams)
+  getMany = (type, ids, options = {}) => {
+    const idsToQuery = ids.slice().map(String)
+    const records = this.getAll(type, options)
 
-    // If the cached record is present
-    if (record && record.id) {
-      // Return data
-      return record
-    } else {
-      // Otherwise fetch it from the server
-      return this.fetchOne(type, id, queryParams)
-    }
+    return records.filter((record) => idsToQuery.includes(record.id))
   }
 
   /**
-   * Like 'findOne', but with an array ids. If there are instances available in the store,
-   * it will return those, otherwise it will trigger a fetch
+   * Fetch all records with the given `type` and `ids` from the server.
+   *
+   * @method fetchMany
+   * @param {String} type the type to get
+   * @param {String} ids the ids of the records to get
+   * @param {Object} options { queryParams }
+   * @return {Array} array of records
+   */
+  fetchMany = async (type, ids, options = {}) => {
+    let idsToQuery = ids.slice().map(String)
+    const queryParams = options.queryParams || {}
+    queryParams.filter = queryParams.filter || {}
+
+    const baseUrl = this.fetchUrl(type, queryParams)
+    const idQueries = deriveIdQueryStrings(idsToQuery, baseUrl)
+    const queries = idQueries.map((queryIds) => {
+      queryParams.filter.ids = queryIds
+      return this.fetchAll(type, { queryParams })
+    })
+    const results = await Promise.all(queries)
+    return Promise.resolve([].concat(...results))
```

Comment:
I feel like there must be a better way to do this. If I just return 
```javascript
return Promise.all(
    idQueries.map((queryIds) => {
        queryParams.filter.ids = queryIds
        return this.fetchAll(type, { queryParams })
     })
)
 ```
 
 It gives me a nested array [[records from first query...], [records from second query...], ...]

What I want is for this method to always return a Promise (resolved or rejected), and if it's resolved the records should be in one array. This solves it but if anyone knows of a better solution that would be great.